cpp/tests/utilities/column_wrapper.cuh

```diff
@@ -0,0 +1,265 @@
+        return static_cast<ColumnType const*>(lhs_col.data)[row] ==
+               static_cast<ColumnType const*>(rhs_col.data)[row];
+      }
+
+      // If one value is valid but the other is not
+      if (lhs_is_valid xor rhs_is_valid) {
+        return false;
+      }
+
+      // If both values are invalid, return true
+      return true;
+    }
+  };
+
+  /**---------------------------------------------------------------------------*
+   * @brief Compares if another column_wrapper is equal to this wrapper.
+   *
+   * Treats NULL == NULL
+   *
+   * @param rhs  The other column_wrapper to check for equality
+   * @return true The two columns are equal
+   * @return false The two columns are not equal
+   *---------------------------------------------------------------------------**/
+  bool operator==(column_wrapper<ColumnType> const& rhs) const {
+    if (the_column.size != rhs.the_column.size) return false;
+    if (the_column.dtype != rhs.the_column.dtype) return false;
+    if (the_column.null_count != rhs.the_column.null_count) return false;
+    if (the_column.dtype_info.time_unit != rhs.the_column.dtype_info.time_unit)
+      return false;
+
+    if (!(the_column.data && rhs.the_column.data))
+      return false;  // if one is null but not both
+
+    if (not thrust::all_of(thrust::make_counting_iterator(0),
+                           thrust::make_counting_iterator(the_column.size),
+                           elements_equal{the_column, rhs.the_column})) {
+      return false;
+    }
+
+    CUDA_RT_CALL(cudaPeekAtLastError());
+
+    return true;
+  }
+
+ private:
+  void initialize_with_host_data(
+      std::vector<ColumnType> const& host_data,
+      std::vector<gdf_valid_type> const& host_bitmask =
+          std::vector<gdf_valid_type>{}) {
+    // Allocate device storage for gdf_column and copy contents from host_data
+    RMM_ALLOC(&(the_column.data), host_data.size() * sizeof(ColumnType), 0);
+    CUDA_RT_CALL(cudaMemcpy(the_column.data, host_data.data(),
+                            host_data.size() * sizeof(ColumnType),
+                            cudaMemcpyHostToDevice));
+
+    // Fill the gdf_column members
+    the_column.size = host_data.size();
+    the_column.dtype = cudf::type_to_gdf_dtype<ColumnType>::value;
+    gdf_dtype_extra_info extra_info;
+    extra_info.time_unit = TIME_UNIT_NONE;
+    the_column.dtype_info = extra_info;
+
+    // If a validity bitmask vector was passed in, allocate device storage
+    // and copy its contents from the host vector
+    if (host_bitmask.size() > 0) {
+      RMM_ALLOC(&(the_column.valid),
+                host_bitmask.size() * sizeof(gdf_valid_type), 0);
+      CUDA_RT_CALL(cudaMemcpy(the_column.valid, host_bitmask.data(),
+                              host_bitmask.size() * sizeof(gdf_valid_type),
+                              cudaMemcpyHostToDevice));
+    } else {
+      the_column.valid = nullptr;
+    }
+    set_null_count(&the_column);
```

Comment:
This is not defined anywhere. You must've missed staging something.

Reply:
Indeed, it's in https://github.com/rapidsai/cudf/pull/577. I'll move it to this PR.

Reply:
I kind of agree with devavret here. I'm uncomfortable with cross-PR dependencies where you can't test A without B and can't test B without A. I think in such situations it should be a single PR.

Reply:
I get it, but I'm trying to avoid monolithic PRs that are hard to review.

This functionality is technically orthogonal to https://github.com/rapidsai/cudf/pull/577, so I opened another PR for it. Note that #694 is already merged into #577.

Reply:
To be clear, #694 PR has no dependency on #577. The `set_null_count` was an omission that I've corrected.